src/t_set.c

```diff
@@ -804,16 +803,18 @@ void spopWithCountCommand(client *c) {
         return;
     }
 
-    /* Case 2 and 3 require to replicate SPOP as a set of SREM commands.
+    /* Case 2 and 3 require to replicate SPOP as a SREM command.
      * Prepare our replication argument vector. Also send the array length
      * which is common to both the code paths. */
-    robj *propargv[3];
+    unsigned long batchsize = 1024;
```

Comment:
set `batchsize` with `count > 1024 ? 1024 : count`, and then we can reuse the `batchsize` to malloc, and don't need `batchindex`.